OpenNefia.Content/UI/UiGroup.cs

```diff
@@ -0,0 +1,212 @@
+                AddChild(SelectedLayer);
+                EntitySystem.InjectDependencies(SelectedLayer);
+                layer.Initialize(SelectedArgs);
+                SetSize(Width, Height);
+                SetPosition(X, Y);
+                SelectedLayer.OnKeyBindDown += OnKeyDown;
+                SelectedLayer.TabEnter();
+                SelectedLayer.GrabFocus();
+            }
+        }
+
+        protected virtual (IAssetDrawable Elem, Vector2i Offset) GetIcon(TLayerArgs args) => (default!, default);
+
+        protected virtual string GetText(TLayerArgs args) => string.Empty;
+
+        public override void SetSize(int width, int height)
+        {
+            base.SetSize(width, height);
+            TabWindow.SetSize(ExtraTabWindowWidth + (Layers.Count * ItemWidth), 24);
+            if (SelectedLayer != null)
+                SelectedLayer.SetPreferredSize();
+        }
+
+        public override void SetPosition(int x, int y)
+        {
+            base.SetPosition(x, y);
+            TabWindow.SetPosition(Width - TabWindow.Width - 20, 32);
+            int tabInfoX = TabWindow.X + (ItemWidth / 2);
+            int tabInfoY = TabWindow.Y + 5;
+            foreach (var layer in Layers)
+            {
+                var iconOffset = Icons[layer.Key].Offset;
+                Icons[layer.Key].Elem.SetPosition(tabInfoX + iconOffset.X, tabInfoY + iconOffset.Y);
+                Texts[layer.Key].SetPosition(tabInfoX, tabInfoY);
+                tabInfoX += ItemWidth;
+            }
+            if (SelectedLayer != null)
+            {
+                UiUtils.GetCenteredParams(SelectedLayer.Width, SelectedLayer.Height, out var bounds);
+                SelectedLayer.SetPosition(bounds.Left, bounds.Top);
+            }
+        }
+
+        public override void Draw()
+        {
+            base.Draw();
+            TabWindow.Draw();
+            TabDeco.Draw(TabWindow.X - 32, TabWindow.Y - 7);
+            foreach(var layer in Layers)
+            {
+                var color = (layer.Value == SelectedLayer) ? Color.White : Color.Gray;
+                Icons[layer.Key].Elem.Color = color;
+                Icons[layer.Key].Elem.Draw();
+                Texts[layer.Key].Color = color;
+                Texts[layer.Key].Draw();
+            }
+            GraphicsEx.SetColor(Color.White);
+            if (SelectedLayer != null)
+                SelectedLayer.Draw();
+        }
+
+        public override void Update(float dt)
+        {
+            base.Update(dt);
+            TabWindow.Update(dt);
+            if (SelectedLayer != null)
+                SelectedLayer.Update(dt);
+        }
+
+        public override UiResult<TResult>? GetResult()
+        {
+            var res = SelectedLayer?.GetResult();
+            if (res != null)
+                SelectedLayer?.TabExit();
+            return res;
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
```

Comment:
Could remove this if it's unchanged from the base, or dispose all child layers here.

Reply:
changed in 9eb449f